Update DSCPullServerSetup help and readme

jpogran opened the pull request:
This PR updates the main DSCPullServerSetup readme to read easier, then updates the PowerShell comment based help for each function to follow normal help standards.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359)
<!-- Reviewable:end -->

msftclas commented:
@jpogran,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by Microsoft. We will now review your pull request.
_Thanks,
Microsoft Pull Request Bot_

kwirkykat commented on DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1:
```diff
@@ -1,34 +1,11 @@
-<#
```
Thank you for updating the function help comment! But can you please do not move the function help comment into the function? It's actually part of our [DSC Resource Kit style guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#functions-have-comment-based-help) that it comes before the function.

This comment goes for the rest of the new function help comments here as well.

Otherwise this PR looks great 😄

johlju commented:
@jpogran are you able to continue the work on this PR? Happy to help you get this reviewed. Could you please rebase this PR against dev so that we get the latest changes?

jpogran commented:
Apologies for the delay @kwirkykat, @johlju, the notification was missed. I rebased this PR against dev and fixed the merge conflicts. I addressed @kwirkykat's comments to the best that I could by moving the help blocks back to just before the function declaration. There are still some whitespace differences, if those are really unacceptable I'll go back and clean those up. Some comments were moved to the help blocks, so those show up as deleted.

johlju commented:
No worries, sorry that nobody sent the question earlier. 

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 2 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCQ6Xr7hVZHOCkUvEd:-LCCQ6Xr7hVZHOCkUvEe:b-2363vl) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L2)):*
> ```
> .Synopsis
> ```

.SYNOPSIS

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 59 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCSSVQBhZwLRocEFV_:-LCCSSVR9MA0KXjPikAc:b-8uhd0q) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L59)):*
> ```
> Function to package modules using a given folder after installing to psmodule path.
> ```

"Function to" seems redundant. Maybe 'Package modules using a given folder after installing to a PowerShell module path ($env:PSModulePath).'

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 63 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCTNU75yE8u0UZTQ9O:-LCCTNU75yE8u0UZTQ9P:boy805q) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L63)):*
> ```PowerShell
> function CreateZipFromPSModulePath
> {
>     param($ListModuleNames, $Destination)
> ```

Can we add these parameters to the comment-based help? If you can't think of a good description, then write 'Description not available.' so that someone else can improve this later.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 102 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCTgkm9lOBKinglnj0:-LCCTgkm9lOBKinglnj1:boy805q) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L102)):*
> ```PowerShell
> function CreateZipFromSource
> {
>     param($Source, $Destination)
> ```

Can we add these parameters to the comment-based help? If you can't think of a good description, then write 'Description not available.' so that someone else can improve this later.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 134 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCThWQ7NAg4Dw2AnyO:-LCCThWQ7NAg4Dw2AnyP:boy805q) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L134)):*
> ```PowerShell
> function PublishModulesAndChecksum
> {
>     param($Source)
> ```

Can we add these parameters to the comment-based help? If you can't think of a good description, then write 'Description not available.' so that someone else can improve this later.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 155 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCTiWt4oMMN4bSEgTn:-LCCTiWt4oMMN4bSEgTo:boy805q) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L155)):*
> ```PowerShell
> function PublishMofDocuments
> {
>    param($Source)
> ```

Can we add these parameters to the comment-based help? If you can't think of a good description, then write 'Description not available.' so that someone else can improve this later.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 193 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCRnOy2SAMaOxdR7C6:-LCCRnOy2SAMaOxdR7C7:byoy86n) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L193)):*
> ```PowerShell
> .PARAMETER Version
>     This is the version of the module
> .PARAMETER PullServerWebConfig
> ```

Can we maybe at least add 'Defaults to $env:SystemDrive\inetpub\wwwroot\PSDSCPullServer\web.config'?

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 194 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCRzmE93_lLuB7mwZC:-LCCRzmE93_lLuB7mwZD:b-ffojpv) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L194)):*
> ```PowerShell
>     This is the version of the module
> .PARAMETER PullServerWebConfig
> .PARAMETER OutputFolderPath
> ```

Can we maybe at least add 'Defaults to $null'? 

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 270 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCRYs99FljqEocyzEB:-LCCRYs99FljqEocyzEC:bmm1lnw) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L270)):*
> ```
> Name
> ```

FullName

---

*[DSCPullServerSetup/README.md, line 3 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCVoFf0jbvLDRFKSkc:-LCCVoFf0jbvLDRFKSkd:b-7048eh) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/README.md#L3)):*
> ```
> DSC Resource
> ```

I think this should be 'DSC module', we are publishing a module with resources, not individual resources?

---

*[DSCPullServerSetup/README.md, line 7 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCVdV8Aa9o7Slhokzd:-LCCVdV8Aa9o7Slhokze:b4s7ywb) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/README.md#L7)):*
> ```
> package DSC Resources
> ```

Maybe 'package a module containing DSC resources'?

---

*[DSCPullServerSetup/README.md, line 11 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCWVh48tSctchtYR2i:-LCCWVh48tSctchtYR2j:biru7oh) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/README.md#L11)):*
> ```
> force the
> ```

I liked 'force packaging the' - it explains more what happens, what do you think?

---

*[DSCPullServerSetup/README.md, line 12 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCWiPR1b_Y6IY5OLxf:-LCCWiPR1b_Y6IY5OLxg:bhsk2oc) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/README.md#L12)):*
> ```
> Resource
> ```

Change to 'module'? see previous comment.

---

*[DSCPullServerSetup/README.md, line 16 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCTxU3B5HYev4BeOzF:-LCCTxU3B5HYev4BeOzG:box1pp3) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/README.md#L16)):*
> ```
> Publish-DSCModuleAndMOF
> ```

The function name is `Publish-DSCModuleAndMof` so I suggest we keep that (lower-case 'of' on 'Mof')

---

*[DSCPullServerSetup/README.md, line 20 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCUCdS2fa5PNcadYyz:-LCCUCdS2fa5PNcadYz-:box1pp3) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/README.md#L20)):*
> ```
> Publish-DSCModuleAndMOF
> ```

The function name is `Publish-DSCModuleAndMof` so I suggest we keep that (lower-case 'of' on 'Mof')

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-:-LCCQFSMACIB43srs0Ff:b-eaeooh)*
<!-- Sent from Reviewable.io -->

johlju commented:
@jpogran Thanks a lot for updating this!

jpogran commented:
@johlju Updated as per comments in reviewable

johlju commented:
@jpogran Could you please go into Reviewable and write 'Done' (or click the done-button) on each of the review comments. This allows me to acknowledge (resolve) the review comments and I know that you are done with each change. After you written 'Done' on each review comment,  or comment on the review comment if it needs further discussion, then click the big green 'Publish' button on top of the page. That will send all your comments back as one big comment to this PR.
You find Reviewable if you click on the big purple Reviewable-button in the PR description.

jpogran commented:
Review status: 0 of 2 files reviewed at latest revision, 15 unresolved discussions, some commit checks failed.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 2 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCQ6Xr7hVZHOCkUvEd:-LD7B4h5DchgClZdDluv:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L2)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> .Synopsis
> ```

.SYNOPSIS
</blockquote></details>

Done.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 59 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCSSVQBhZwLRocEFV_:-LD7B7hf2AgWCAwmRwwc:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L59)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> Function to package modules using a given folder after installing to psmodule path.
> ```

"Function to" seems redundant. Maybe 'Package modules using a given folder after installing to a PowerShell module path ($env:PSModulePath).'
</blockquote></details>

Done.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 63 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCTNU75yE8u0UZTQ9O:-LD7BH-X2e5Liy3fqYh5:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L63)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Can we add these parameters to the comment-based help? If you can't think of a good description, then write 'Description not available.' so that someone else can improve this later.
</blockquote></details>

Done.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 102 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCTgkm9lOBKinglnj0:-LD7BJTf1eeU50Kwr98j:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L102)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Can we add these parameters to the comment-based help? If you can't think of a good description, then write 'Description not available.' so that someone else can improve this later.
</blockquote></details>

Done.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 134 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCThWQ7NAg4Dw2AnyO:-LD7BLax-KFCwm8pGcnt:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L134)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Can we add these parameters to the comment-based help? If you can't think of a good description, then write 'Description not available.' so that someone else can improve this later.
</blockquote></details>

Done.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 155 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCTiWt4oMMN4bSEgTn:-LD7BMS86JwPVFnTm2er:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L155)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Can we add these parameters to the comment-based help? If you can't think of a good description, then write 'Description not available.' so that someone else can improve this later.
</blockquote></details>

Done.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 193 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCRnOy2SAMaOxdR7C6:-LD7BNotDlSX9PK0mcvc:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L193)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Can we maybe at least add 'Defaults to $env:SystemDrive\inetpub\wwwroot\PSDSCPullServer\web.config'?
</blockquote></details>

Done.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 194 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCRzmE93_lLuB7mwZC:-LD7BOex1SvfcAglZ6zU:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L194)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Can we maybe at least add 'Defaults to $null'? 
</blockquote></details>

Done.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 270 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCRYs99FljqEocyzEB:-LD7BQk3E8ovpeq9kPYk:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L270)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> Name
> ```

FullName
</blockquote></details>

Done.

---

*[DSCPullServerSetup/README.md, line 3 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCVoFf0jbvLDRFKSkc:-LD7BRbC7qc1JaWHdHTe:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/README.md#L3)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> DSC Resource
> ```

I think this should be 'DSC module', we are publishing a module with resources, not individual resources?
</blockquote></details>

Done.

---

*[DSCPullServerSetup/README.md, line 7 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCVdV8Aa9o7Slhokzd:-LD7BSQ42vG5NsGQkOWD:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/README.md#L7)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> package DSC Resources
> ```

Maybe 'package a module containing DSC resources'?
</blockquote></details>

Done.

---

*[DSCPullServerSetup/README.md, line 11 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCWVh48tSctchtYR2i:-LD7Bo0c0e26euz0bd9L:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/README.md#L11)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> force the
> ```

I liked 'force packaging the' - it explains more what happens, what do you think?
</blockquote></details>

Done.

---

*[DSCPullServerSetup/README.md, line 12 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCWiPR1b_Y6IY5OLxf:-LD7C4K134nNPxxIE1tV:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/README.md#L12)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> Resource
> ```

Change to 'module'? see previous comment.
</blockquote></details>

Done.

---

*[DSCPullServerSetup/README.md, line 16 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCTxU3B5HYev4BeOzF:-LD7BrUd1sxNs6S2Ijur:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/README.md#L16)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> Publish-DSCModuleAndMOF
> ```

The function name is `Publish-DSCModuleAndMof` so I suggest we keep that (lower-case 'of' on 'Mof')
</blockquote></details>

Done.

---

*[DSCPullServerSetup/README.md, line 20 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCUCdS2fa5PNcadYyz:-LD7BsYYCrCsr26HvIiN:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/README.md#L20)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> Publish-DSCModuleAndMOF
> ```

The function name is `Publish-DSCModuleAndMof` so I suggest we keep that (lower-case 'of' on 'Mof')
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359)*
<!-- Sent from Reviewable.io -->

jpogran commented:
@johlju I think I did as you asked, first time using reviewable.

johlju commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LD7Tdpp2Pz9F7j2NV0o:-LD7Tdpq5VjF1jgWusa1:b-v7buj) (no related file):*
Could you please add an descriptive entry, for each change/issue, to the Unreleased section of the change log in the file README.md? If an entry resolves an issue please reference the issue.

You may (this is optional) suffix each entry with your name. Example of the format for the entry
```
* Descriptive entry ([issue #123](https://github.com/PowerShell/xPSDesiredStateConfiguration/issues/123). [Name/Alias (@github_account)](https://github.com/github_account)`
```

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359)*
<!-- Sent from Reviewable.io -->

johlju commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 24 at r2](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LD7VpT0AkqeyZ46iqBo:-LD7VpT0AkqeyZ46iqBp:b-dp4l5t) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/1b784836d7687c227fbd68eb577f60fca1e70997/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L24)):*
> ```PowerShell
> #>
> 
> # Tools to use to package DSC modules and mof configuration document and publish them on enterprise DSC pull server in the required format
> ```

This comment should be removed, and the blank row above it, so the comment based help is directly over the function-block.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359)*
<!-- Sent from Reviewable.io -->

johlju commented:
@jpogran You need to rebase this PR too. Please look here how to do that; https://github.com/PowerShell/DscResources/blob/master/GettingStartedWithGitHub.md#resolve-merge-conflicts
Let me know if you need any assistance.

jpogran commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LD7Tdpp2Pz9F7j2NV0o:-LD7x5906PAYvOwGAisC:b-4mql89) (no related file):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Could you please add an descriptive entry, for each change/issue, to the Unreleased section of the change log in the file README.md? If an entry resolves an issue please reference the issue.

You may (this is optional) suffix each entry with your name. Example of the format for the entry
```
* Descriptive entry ([issue #123](https://github.com/PowerShell/xPSDesiredStateConfiguration/issues/123). [Name/Alias (@github_account)](https://github.com/github_account)`
```
</blockquote></details>

There's no issue created alongside this PR, it was done before there were such standards. Do I need to create one to satisfy this requirement now?

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 2 at r1](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LCCQ6Xr7hVZHOCkUvEd:-LD7xe743Xt7PJhjAKtP:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/de4ef8b3aec5546bd22d9a2dcbfd822fe5d48973/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L2)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Sorry if I was unclear. :/ This should be `.SYNOPSIS` with all upper case.
</blockquote></details>

Done.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 24 at r2](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LD7VpT0AkqeyZ46iqBo:-LD82FXG3VnRyvOYQkMz:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/2a6f4e4996ae5e83d07e2edb7c2d530abbfe64af/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L24)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This comment should be removed, and the blank row above it, so the comment based help is directly over the function-block.
</blockquote></details>

Done.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 102 at r2](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LD7RyTF8_ctZMSWAhsY:-LD836SR4aQj8mDmpHdg:b-4ttphg) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/2a6f4e4996ae5e83d07e2edb7c2d530abbfe64af/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L102)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This description seems wrong, it creates zip's the source? Maybe the comment on line 111 can be moved here as the description?
</blockquote></details>

Tried a slightly different explanation

---

*[DSCPullServerSetup/README.md, line 3 at r2](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LD7Sg9rBkTScMMi1fYj:-LD83Va92vq31G06OaDW:b-h81w82) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/2a6f4e4996ae5e83d07e2edb7c2d530abbfe64af/DSCPullServerSetup/README.md#L3)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> Module
> ```

Maybe we should write 'module' with lower-case 'm'? 
</blockquote></details>

Done. Also fixed typo in deployment.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reviewed 2 of 2 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LD7Tdpp2Pz9F7j2NV0o:-LDAkYMQDk5kPl8RjyPs:b-d52n51) (no related file):*
<details><summary><i>Previously, jpogran (James Pogran) wrote…</i></summary><blockquote>

There's no issue created alongside this PR, it was done before there were such standards. Do I need to create one to satisfy this requirement now?
</blockquote></details>

No need to create an issue. Leave out the issue reference part if there are no issues that are closed. *Normally an issue is opened for the community to discuss a change, but that is not a hard rule, depends on the change. :slightly_smiling_face:*

---

*[README.md, line 643 at r4](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LDArNkhDHQM8y5z_uWM:-LDArNkhDHQM8y5z_uWN:bebbmb8) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/09bc8cd98717540f1f6dbf63c166c26948117883/README.md#L643)):*
> ```
> `
> ```

Please remove the starting and ending ` around the name.  Removing this will allow the name to link to your profile page.

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 102 at r2](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LD7RyTF8_ctZMSWAhsY:-LDAlacnAdd8MRiGgE7M:bc90rl5) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/2a6f4e4996ae5e83d07e2edb7c2d530abbfe64af/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L102)):*
<details><summary><i>Previously, jpogran (James Pogran) wrote…</i></summary><blockquote>

Tried a slightly different explanation
</blockquote></details>

The description seems slightly wrong still :thinking: Also `$env:PSModulePath` is more than one path.

Maybe this?

"Deploys all DSC resource modules in the path assigned to the parameter Source. The DSC resource modules are copied to the path '$env:ProgramFiles\WindowsPowerShell\Modules', and also packaged into a zip archive that is saved to the path assigned to the parameter Destination."

---

*[DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1, line 57 at r4](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-LDAo_a44KQjn7Ct_LAJ:-LDAo_a44KQjn7Ct_LAK:b-sepn9c) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/09bc8cd98717540f1f6dbf63c166c26948117883/DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1#L57)):*
> ```PowerShell
> <#
> .SYNOPSIS
>     Package modules using a given folder after installing to a PowerShell module path ($env:PSModulePath).
> ```

When looking at the below SYNOPSIS, this description seemed off to. 

Maybe this instead?

"Creates a zip archive containing all the modules whose module name was assigned to the parameter ListModuleNames. The zip archive is created in the path assigned to the parameter Destination."

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359)*
<!-- Sent from Reviewable.io -->

johlju commented:
Great work! :smiley: Looking good, could you just rebase again? I just merged another PR so this one is conflicting on the change log.

---

Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-:-LDH20TvCBH6_2xPr4Px:b-kuaepc)*
<!-- Sent from Reviewable.io -->

jpogran commented:
No problem. Rebased

johlju commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/359#-:-LDMdaz11nxzK1mLtZnH:bnfp4nl)*
<!-- Sent from Reviewable.io -->

johlju commented:
The tests failed on something else that is not part of this PR. Closing and reopening this PR to kick off the tests.
